src/library/scala/collection/immutable/LazyList.scala

```diff
@@ -1370,7 +1377,7 @@ object LazyList extends SeqFactory[LazyList] {
         case a => init += a.asInstanceOf[A]
       }
       val tail = in.readObject().asInstanceOf[LazyList[A]]
```

Comment:
does this even... work? I can't comment on the `writeObject` method as it's too far from the changes, but wouldn't it just re-run through a new proxy? or is it specified somewhere that if you try to serialize the type of a proxy from within a proxy it will use default serialization?

Reply:
Note that there's a serialization proxy. The enclosing `readObject` is in the `SerializationProxy`, so `in.readObject` here will run the default Java deserialization on a LazyList instance.